### PR TITLE
get post-sync absorb out of the sync handler and monitor the worker

### PR DESCRIPTION
**Pull request by evanmcc**

### CODE REVIEW

**Comment by Vagabond** [comment]
I think this is working now.

**Comment by Vagabond** [comment]
This is ready for review @evanmcc @madninja

File: src/blockchain_worker.erl [hunk 1]
```diff
@@ -414,6 +477,42 @@ handle_info({'DOWN', GossipRef, process, _GossipPid, _Reason},
                                     {blockchain_gossip_handler, [Swarm, Blockchain]}),
     NewGossipRef = erlang:monitor(process, Gossip),
     {noreply, State#state{gossip_ref = NewGossipRef}};
+handle_info({'DOWN', AbsorbRef, process, AbsorbPid, Reason},
+            #state{absorb_info = {AbsorbPid, AbsorbRef}, absorb_retries = Retries} = State) ->
+    case Reason of
+        normal ->
+            lager:info("Absorb process completed successfully"),
+            {noreply, State#state{sync_paused=false, absorb_info=undefined}};
```
Comment (evanmcc) [inline src/blockchain_worker.erl:485]: I think that we need to resume the sync timer here?

File: src/blockchain_worker.erl [hunk 2]
```diff
@@ -414,6 +477,42 @@ handle_info({'DOWN', GossipRef, process, _GossipPid, _Reason},
...
+            %% ran out of retries
+            {stop, Reason, State}
+    end;
+handle_info({'DOWN', ResyncRef, process, ResyncPid, Reason},
+            #state{resync_info = {ResyncPid, ResyncRef}, resync_retries = Retries} = State) ->
+    case Reason of
+        normal ->
+            lager:info("Resync process completed successfully"),
+            %% check if we have any pending assume valids to take care of
+            blockchain:init_assumed_valid(State#state.blockchain, get_assume_valid_height_and_hash()),
```
Comment (evanmcc) [inline src/blockchain_worker.erl:503]: maybe need to conditionally reinit the sync timer here?

**Pull request merged**